Homework 3/script.js

```diff
@@ -39,12 +51,56 @@ $( document ).ready(function() {
 	});
 
 	$("#signUpButton").click(function() {
+		$("#signUpErrorMessage").hide();
+		$("#signUpErrorMessage").empty();
+		if(($('#signUpPasswordInput').val() != '') && ($('#signUpPasswordConfirmInput').val() != '') && ($('#signUpEmailInput').val() != '') && ($('#signUpUsernameInput').val() != '')){
+			if($('#signUpPasswordInput').val() == $('#signUpPasswordConfirmInput').val()){
+				if($('input[name=accountType]:checked').val() == "storeOwner"){
+					if(($('#signUpStoreNameInput').val() != '') && ($('#signUpStoreImageUrlInput').val() != '')){
+						productManager.createUser($('#signUpUsernameInput').val(), $('#signUpEmailInput').val(), $('#signUpPasswordInput').val(), true, function(success, createdUser){
+							if(success){
+								productManager.createStore($('#signUpStoreNameInput').val(), $('#signUpStoreImageUrlInput').val(), ($('input[name=isOpen]:checked').val() == "true"), function(store){
+									user = new StoreOwner(createdUser, store);
+									productManager.createStoreUserRelation(store.cloudboostObject, createdUser, function(){
+										loadMallPage();
+									})
+								});
+							}
+						});
+					}
+					else{
+						$("#signUpErrorMessage").show();
+						$("#signUpErrorMessage").append("* You must fill in all of the fields.");
+					}
+				}
+				else{
+					productManager.createUser($('#signUpUsernameInput').val(), $('#signUpEmailInput').val(), $('#signUpPasswordInput').val(), false, function(success, createdUser){
+						if(success){
+							user = createdUser;
+							loadMallPage();
+						}
+					});
+				}
+			}
+			else{
+				$("#signUpErrorMessage").show();
```

Comment:
Same error message and function is showing up over and over. Copy and paste might make sense now, but what if I told you to refactor the `else` behavior? Now you have to copy and paste everywhere. If the logic gets more complicated, you'll end up spending more time looking for the places to update.